src/components/vm/vmActions.tsx

```diff
@@ -561,4 +519,43 @@ const VmActions = ({
     );
 };
 
+export const VmRestartDialog = ({ vm } : { vm: VM }) => {
+    async function onRestart(force: boolean) {
+        startOperationProgress(vm);
+        store.dispatch(
+            updateVm({
+                connectionName: vm.connectionName,
+                name: vm.name,
+                onShutOff: vmStart,
+            })
+        );
+        try {
+            await vmDomainMethod<void>(vm, force ? 'Destroy' : 'Shutdown', 'u', 0);
+        } catch (ex) {
+            setVmError(vm, cockpit.format(_("VM $0 failed to shutdown"), vm.name), ex);
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2376-46b24e73-20251028-114421-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)